src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -135,6 +135,10 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 	private _shouldSyncWhenReady: boolean = false;
 	private _suggestTelemetry: TerminalSuggestTelemetry | undefined;
 
+	private _completionRequestTimestamp: number | undefined;
+	private _completionLatency: number | undefined;
+	private _hasShownCompletions: boolean = false;
```

Comment:
just realized this won't work because tracked per terminal, fixing

Reply:
will use the storage service and reset on window reload?

Reply:
oh, this should probably be per shell type?

Reply:
It's bound to the lifecycle of the extension host. I think you can track this in a top level object (not inside the class). Then listen to an extensions service for when the extensions resets and clear it out, shouldn't matter that it will cause that to clear that multiple times.